cfme/tests/cloud_infra_common/test_vm_instance_analysis.py

```diff
@@ -182,27 +172,23 @@ def set_agent_creds(appliance, request, provider):
 
     @request.addfinalizer
     def _remove_docker_creds():
-        view = navigate_to(provider, "Edit")
+        view = navigate_to(ssa_provider, "Edit")
         ssa_agent_view = view.browser.create_view(EC2EndpointForm)
         ssa_agent_view.smart_state_docker.fill({'username': ""})
         ssa_agent_view.smart_state_docker.change_pass.click()
         view.save.click()
 
 
-@pytest.fixture(scope="module")
-def local_setup_provider(request, setup_provider_modscope, provider, appliance):
-
-    # TODO: allow for vddk parameterization
-    if provider.one_of(VMwareProvider):
-        appliance.install_vddk()
-        request.addfinalizer(appliance.uninstall_vddk)
-        set_hosts_credentials(appliance, request, provider)
-
-    if provider.one_of(EC2Provider):
-        set_agent_creds(appliance, request, provider)
+def set_host_creds(request, appliance, ssa_provider, ssa_vm):
+    host = ssa_vm.host
+    host_data = [data for data in ssa_provider.data['hosts'] if data['name'] == host.name]
+    host.update_credentials_rest(credentials=[h.get('credentials') for h in host_data][0])
```

Comment:
It's not good iteration...
I would go like this
`host_data = [data for data in ssa_provider.data['hosts'] if data['name'] == host.name][0]` as we know data for perticular host else I will rename it as `hosts_data` in case multiple host possible with same name.

after that
`host_data.get("credentials")`